client/homebrew/editor/stringArrayEditor/stringArrayEditor.jsx

```diff
@@ -0,0 +1,138 @@
+const React = require('react');
+const createClass = require('create-react-class');
+const _     = require('lodash');
+
+const StringArrayEditor = createClass({
+	displayName     : 'StringArrayEditor',
+	getDefaultProps : function() {
+		return {
+			label         : '',
+			values        : [],
+			valuePatterns : null,
+			placeholder   : '',
+			unique        : false,
+			cannotEdit    : [],
+			onChange      : ()=>{}
+		};
+	},
+
+	getInitialState : function() {
+		return {
+			valueContext : !!this.props.values ? this.props.values.map((value)=>({
+				value,
+				editing : false
+			})) : [],
+			temporaryValue : '',
+			updateValue    : ''
+		};
+	},
+
+	componentDidUpdate : function(prevProps) {
+		if(!_.eq(this.props, prevProps)) {
+			this.setState({
+				valueContext : !!this.props.values ? this.props.values.map((newValue)=>({
```

Comment:
I don't think Javascript needs `!!` to typecast to boolean. `this.props.values` will already be evaluated as a boolean because of the ternary operator.